src/MD5_fmt.c

```diff
@@ -157,96 +157,10 @@ static void done(void)
 	MEM_FREE(saved_key);
 }
 
-static int get_hash_0(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_0;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_0;
-#endif
-}
-
-static int get_hash_1(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_1;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_1;
-#endif
-}
-
-static int get_hash_2(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_2;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_2;
-#endif
-}
-
-static int get_hash_3(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_3;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_3;
-#endif
-}
-
-static int get_hash_4(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_4;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_4;
-#endif
-}
-
-static int get_hash_5(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_5;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_5;
-#endif
-}
-
-static int get_hash_6(int index)
-{
-#ifdef SIMD_PARA_MD5
-	unsigned int x,y;
-	x = index&(SIMD_COEF_32-1);
-	y = (unsigned int)index/SIMD_COEF_32;
-	return ((MD5_word *)sout)[x+y*SIMD_COEF_32*4] & PH_MASK_6;
-#else
-	init_t();
-	return MD5_out[index][0] & PH_MASK_6;
-#endif
-}
+#define COMMON_GET_HASH_SIMD32 4
+#define COMMON_GET_HASH_VAR MD5_out
+#define COMMON_GET_HASH_SIMD_VAR sout
+#include "common-get-hash.h"
```

Comment:
Are you sure this works right in non-SIMD OpenMP builds? You're dropping `init_t()` here.

Reply:
Let me check.  If so, then it can be backed out (with a comment), and left in original state.

Reply:
Initial testing, nope. MD5_fmt.c will not build with this change (missing 't' variable in other code), so you are correct, this is NOT a candidate which can be done in this fashion.  I missed this, because my non-SIMD testing was using Makefile.legacy using win32-cygwin-x86-any target, and it did not have openmp enabled.  Once I enabled, that build failed ASAP!  VERY good catch @solardiz

Reply:
dd87fbf9d reverts MD5_fmt.c  If the 't' variable is required (it sounds like it), then this format can not be handled in a 'generic-include' manner.

Reply:
I am going to add another target into Makefile.legacy so I can build with or without openmp on non-SIMD builds.  (but not on this PR, it does not belong here)

Reply:
BTW, do we make any OpenMP non-SIMD builds in CI? We should. Didn't CI report all tests passed prior to you dropping the md5crypt changes? If so, that's a problem with our CI tests we should look into.

Reply:
It appears to me that on PR's only Travis is doing builds.  AFAIK, only CircleCI tests non-SIMD builds. I made this statement already (on the PR for the GETPOS/[sg]et_key generic header logic).  That one is just as important to test well on non-simd builds.  I will check out that PR, and do a build, now that I have a non-SIMD-gomp target, and see if it builds, AND see if it tests clean.

BUT YES, your point is 100% valid, and I have made it also.

Now, on 'real' commits, the CI's tested Are both CircleCI and TravisCI. But the PR builds do not seem so (at least the PRs I have started pushing)   It may be that since the code originated from my repo, that it does not build on circle, I am not sure.

Reply:
#2955 added point out this issue.

Reply:
> It may be that since the code originated from my repo, that it does not build on circle.

Even for PRs Circle follows your repo configuration. And your builds are disabled on Circle (and Travis, but it is smarter).

Reply:
Ok, the magic question asking for the secret handshake.  How do I enable Circle (and Travis) in my repo ??   I have gone to the circle site, and for jfoug/JohnTheRipper, things 'appear' like they should be following the project.  But no pushes are getting built?    It would be nice to get this setup.  Once done, it should just work.

I also saw config for 'always' test forked PR's.  This sounds like something with the magnumripper/JohnTheRipper repo should be setup to do, so that it simply forces the CI building on a PR, no matter what the underlying PR is setup to do.